flink/src/main/java/org/apache/zeppelin/flink/FlinkInterpreter.java

```diff
@@ -48,18 +48,39 @@
 
   public FlinkInterpreter(Properties properties) {
     super(properties);
-    this.innerIntp = new FlinkScalaInterpreter(getProperties());
+  }
+
+  private String extractScalaVersion() throws InterpreterException {
+    String scalaVersionString = scala.util.Properties.versionString();
+    LOGGER.info("Using Scala: " + scalaVersionString);
+    if (scalaVersionString.contains("version 2.10")) {
+      return "2.10";
+    } else if (scalaVersionString.contains("version 2.11")) {
+      return "2.11";
+    } else if (scalaVersionString.contains("version 2.12")) {
+      return "2.12";
+    } else {
+      throw new InterpreterException("Unsupported scala version: " + scalaVersionString);
+    }
```

Comment:
Do we plan to support multiple Scala versions in the future, and have some functionality depending on it?
Otherwise I would maybe just implement something like "isSupportedScalaVersion" with one check for Scala 2.11 as of now.

Reply:
addressed